### PR TITLE
ES6: add method definition support #40

### PR DESCRIPTION
**Pull request by dwightmulcahy**

### CODE REVIEW

File: test/object-expressions/object-expression.js [hunk 1]
```diff
@@ -218,5 +218,79 @@ suite("Parser", function () {
...
+              new Shift.VariableDeclaration("let",
+                [new Shift.VariableDeclarator(
+                  new Shift.BindingIdentifier(new Shift.Identifier("a"),null),null)
+                ])
+            )]
+          ), false
+        )
+      ])
+    );
+ testParse("({a(b){}})", expr,
```
Comment (michaelficarra) [inline test/object-expressions/object-expression.js]: Indentation.

Comment (dwightmulcahy) [inline test/object-expressions/object-expression.js]: fixed.

File: test/object-expressions/object-expression.js [hunk 2]
```diff
@@ -218,5 +218,79 @@ suite("Parser", function () {
...
     );
+
+    testParse("({a(){}})", expr,
+      new Shift.ObjectExpression([
+        new Shift.Method(
+          false,
+          new Shift.StaticPropertyName("a"),
+          [],
+          null,
+          new Shift.FunctionBody([], []), false
```
Comment (michaelficarra) [inline test/object-expressions/object-expression.js]: What is with the trailing `false` here?

File: test/object-expressions/object-expression.js [hunk 3]
```diff
@@ -218,5 +218,79 @@ suite("Parser", function () {
...
+    testParse("({a(){let a;}})", expr,
+      new Shift.ObjectExpression([
+        new Shift.Method(
+          false,
+          new Shift.StaticPropertyName("a"),
+          [],
+          null,
+          new Shift.FunctionBody(
+            [],
+            [new Shift.VariableDeclarationStatement(
```
Comment (michaelficarra) [inline test/object-expressions/object-expression.js]: This array formatting is inconsistent with the rest of the codebase. Please make it consistent.

File: test/object-expressions/object-expression.js [hunk 4]
```diff
@@ -218,5 +218,79 @@ suite("Parser", function () {
...
+              new Shift.VariableDeclaration("let",
+                [new Shift.VariableDeclarator(
+                  new Shift.BindingIdentifier(new Shift.Identifier("a"), null), null)
```
Comment (michaelficarra) [inline test/object-expressions/object-expression.js]: BindingIdentifier expects a single parameter.

File: test/object-expressions/object-expression.js (hunk 3 above)
Comment (dwightmulcahy) [inline test/object-expressions/object-expression.js]: assuming for multiple values:
         [
           new Shift.BindingIdentifier(new Shift.Identifier("b")),
           new Shift.BindingIdentifier(new Shift.Identifier("c"))
         ],

for a single value that can fit on a line: 
         [new Shift.BindingIdentifier(new Shift.Identifier("b"))],

**Comment by michaelficarra** [comment]
Add a test that includes a rest parameter, then LGTM.

**Comment by dwightmulcahy** [comment]
added the following to test rest parameters... 

testParse("({a(b,...c){}})", expr, ...

File: test/object-expressions/object-expression.js [hunk 5]
```diff
@@ -218,5 +218,94 @@ suite("Parser", function () {
...
+        new Shift.Method(
+          false,
+          new Shift.StaticPropertyName("a"),
+          [new Shift.BindingIdentifier(new Shift.Identifier("b"))],
+          null,
+          new Shift.FunctionBody([], [])
+        )
+      ])
+    );
+   testParse("({a(b,...c){}})", expr,
```
Comment (michaelficarra) [inline test/object-expressions/object-expression.js]: Indentation.

Comment (dwightmulcahy) [inline test/object-expressions/object-expression.js]: DAMN IT!!!  fixed.

**Pull request merged**